### PR TITLE
Dedicated widget for generic permissions

### PR DESCRIPTION
**Pull request by leplatrem**
capture d ecran de 2016-10-13 11-57-01

### CODE REVIEW

**Comment by leplatrem** [comment]
@n1k0 r?

File: src/components/PermissionsForm.js [hunk 1]
```diff
@@ -2,19 +2,19 @@ import React, { Component } from "react";
 import Form from "react-jsonschema-form";
 
 import {
-  permissionsObjectToList,
-  permissionsListToObject,
+  permissionsToFormData,
+  formDataToPermissions,
```
Comment (n1k0) [inline src/components/PermissionsForm.js]: ❤️

File: src/permission.js [hunk 2]
```diff
@@ -85,35 +88,85 @@ export function canEditRecord(session: SessionState, bucket: BucketState, collec
...
-                ...existing,
-                permissions: [...existing.permissions, permissionName]
-              };
-            } else {
-              return perm;
-            }
-          });
-        } else {
-          acc = [...acc, {principal, permissions: [permissionName]}];
+        if (principal === EVERYONE) {
```
Comment (n1k0) [inline src/permission.js]: Nit: call me old fashioned but I could see a `switch` statement here

File: test/permission_test.js [hunk 3]
```diff
@@ -273,38 +273,31 @@ describe("canEditRecord", () => {
 describe("Permission form mappers", () => {
   const permissions = {
     read: ["a", "b", "c"],
-    write: ["a"],
+    write: ["system.Authenticated"],
```
Comment (n1k0) [inline test/permission_test.js]: We may want to import (and so, to expose) the constant here.

**Pull request merged**